server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -912,8 +912,16 @@ private void synchronizeJobSchedule() throws SchedulerException {
 
         // Unschedule dead/invalid jobs
         for (JobKey key : unschedule) {
-            this.scheduler.deleteJob(key);
-            log.info("Removed existing schedule for job: {}", key.getName());
+            List<Trigger> jobTrigger = (List<Trigger>) this.scheduler.getTriggersOfJob(key);
+
+            if (!jobTrigger.isEmpty()) {
+                this.scheduler.unscheduleJobs(jobTrigger
+                    .stream()
+                    .map(Trigger::getKey)
+                    .collect(Collectors.toList()));
+                log.info("Removed existing schedule for job: {}", key.getName());
+            }
```

Comment:
We should have a contingency here in the event jobTriggers is empty. In such a case, we'll have a job key associated with nothing, and that should get cleaned up appropriately.

Reply:
Added case to delete the job for proper cleanup (assuming we might have stale data). 
Although since all Candlepin quartz jobs are non-durable, it is automatically deleted from the scheduler once there are no longer any active triggers associated with it.